backend/src/lib.rs

```diff
@@ -169,12 +58,14 @@ pub enum Error {
     EmptyWitness,
     #[error("the backend has no setup operations")]
     NoSetupAvailable,
-    #[error("proof generation failed")]
-    ProofFailed(String),
-    #[error("proof verification failed")]
-    VerificationFailed(String),
-    #[error("verification key generation failed")]
-    VerificationKeyFailed(String),
+    #[error("backend specific error")]
+    BackendError(String),
```

Comment:
I think we could keep the old errors since they are generic enough for the things we want to do.

Reply:
I changed it because this way it is simpler and less error prone to write the backend implementation: you can  just use `?` to turn return from a `Result<_, String>` and it becomes a backend error. Whether it is a proving or a verification error, it is obvious from the function that was called.

Reply:
Sure, not too strongly opinionated here.